### DISCUSSION TITLE
SQLite tables does not want to be created

### DISCUSSION THREAD

**Discussion by V-ed** [prisma/prisma#5812] [Q&A] [answered]
When running my application / trying to seed the db using `npx prisma db seed --preview-feature` / etc, I often get the error :

```bash
PrismaClientKnownRequestError2 [PrismaClientKnownRequestError]: The table `dev.users` does not exist in the current database.
```

... and this error happens for all of my tables, and also if the sqlite database file exists or not.

The only way I'm able to not have this issue is when trying to run the first migration (with `npx prisma migrate dev --preview-feature`), which passes fine, then I'm able to seed, for example, or run the application.

The issue I have is that I want to be able to run my tests with a test database that may not already exist (in CI, for example), and I am running into this "table not created in this database" for my tests too.

Any idea how could I use Prisma without running an initial migration / creating a test database for unit tests correctly?

(I know that `creating a test database for unit tests correctly` might be the ultimate issue here, but I searched a lot for "Table not created" and found absolutely nothing 😕)

Thanks for the work on Prisma, I jumped from TypeORM to test Mikro Orm, which I ended up to not like it and fell in love with Prisma once I started using it!

**Comment by ryands17** [comment, marked as answer]
@V-ed 👋 

If you're getting this error with testing, I would suggest looking at this repo where I have [setup the test environment](https://github.com/ryands17/nexus-auth/blob/master/prisma/prisma-test-environment.js) and this works fine for every test file.

We also have a similar example [here](https://github.com/prisma/prisma-examples/tree/latest/typescript/testing-express) that has a similar setup. Let me know if this helps :)

**Reply by V-ed** [reply]
Hi @ryands17 !

Looking at the examples you provided, I can understand that I'll need to run either a migration or a `prisma db push [...]` somewhere in my test suite, I'll test that for sure!

Also, I can see that in those tests, the test database is setup when starting all the tests, and one test can modify the database for the other tests, making a single unit test a bit unreliable IMO - in my setup, I have made seeds that are usable in tests (except that I kept getting table not found errors, of course haha), but I was wondering if there was a way to completely drop all data from the database for each test? I think I saw somewhere the ability to drop the database schema, but could that be of help here or will dropping the schema also means I'll have to run a migration / db push again for each test?

I'll test the configurations you gave me as an example and I'll report back here, thanks for the info!

**Reply by V-ed** [reply]
Yup, setting up the database using a node environment and more importantly the ``await exec(`${prismaBinary} db push --preview-feature`);`` line worked wonderfully, thanks for the examples links!

I'll still have to figure out how to clean and seed my db for every test, as I do want clean data when testing, but that'll be another issue so I've marked this as answered. (maybe I'll even go against this for performance reasons? eh, I'll see haha)

Thanks again and have a good day! :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
